app/uk/gov/hmrc/teamsandrepositories/services/JenkinsService.scala

```diff
@@ -47,15 +55,39 @@ class JenkinsService @Inject()(repo: BuildJobRepo, jenkinsConnector: JenkinsConn
       persist <- repo.update(buildJobs)
     } yield persist
 
-  def triggerBuildJob(serviceName:String, url: String,
-                      timestamp: Instant)(implicit ec: ExecutionContext): Future[Unit] = {
+  def triggerBuildJob(serviceName: String, url: String,
+                      timestamp: Instant)(implicit ec: ExecutionContext): Future[JenkinsBuildData] = {
     for {
       latestBuild <- jenkinsConnector.getLastBuildTime(url)
-      _ <- if (latestBuild.timestamp.equals(timestamp)) {
-          logger.info(s"Triggering build for ${serviceName}")
-          jenkinsConnector.triggerBuildJob(url)
-        }
-        else Future.successful[Unit](())
-    } yield ()
+      response <- {
+        logger.info(s"Triggering build for $serviceName")
+        jenkinsConnector.triggerBuildJob(url)
+      } if latestBuild.timestamp.equals(timestamp)
+      queueUrl = s"${response.header("Location").get.replace("http:", "https:")}api/json"
+      queue <- getQueue(queueUrl)
+      build <- getBuild(queue.executable.get.url)
+    } yield build
+  }
+
+  private def getQueue(queueUrl: String)(implicit ec: ExecutionContext): Future[JenkinsQueueData] = {
+    Source.repeat(42)
+        .throttle(1, 10 seconds)
+        .mapAsync(parallelism = 1)(_ => jenkinsConnector.getQueueDetails(queueUrl))
+        .filter(queue => queue.cancelled.nonEmpty)
+      .map(queue => {
+        if (queue.cancelled.get) throw new Exception("Queued Job cancelled")
+        queue
+      })
+        .filter(queue => queue.executable.nonEmpty)
+        .runWith(Sink.head)
+  }
+
+  private def getBuild(buildUrl: String)(implicit ec: ExecutionContext): Future[JenkinsBuildData] = {
+    Source.repeat(42)
+      .throttle(1, 1 minute)
```

Comment:
should these values be in config?
`getQueue` too.

Reply:
Done